plugins/turing/config/config.go

```diff
@@ -31,20 +31,117 @@ type RemoteUI struct {
+	RemoteUI                     RemoteUI                     `json:"remote_ui"`
+	RunnerDefaults               RunnerDefaults               `json:"runner_defaults"`
+	TreatmentServicePluginConfig TreatmentServicePluginConfig `json:"treatment_service_plugin_config"`
 }
 
 // ExperimentRunnerConfig is used to parse the XP runner config during initialization
 type ExperimentRunnerConfig struct {
-	Endpoint          string     `json:"endpoint" validate:"required"`
-	ProjectID         int        `json:"project_id" validate:"required"`
-	Passkey           string     `json:"passkey" validate:"required"`
-	Timeout           string     `json:"timeout" validate:"required"`
-	RequestParameters []Variable `json:"request_parameters" validate:"required,dive"`
+	Endpoint               string                  `json:"endpoint" validate:"required"`
+	ProjectID              int                     `json:"project_id" validate:"required"`
+	Passkey                string                  `json:"passkey" validate:"required"`
+	Timeout                string                  `json:"timeout" validate:"required"`
+	RequestParameters      []Variable              `json:"request_parameters" validate:"required,dive"`
+	TreatmentServiceConfig *TreatmentServiceConfig `json:"treatment_service_config"`
+}
+
+type TreatmentServicePluginConfig struct {
+	Port       int      `json:"port" default:"8080"`
+	ProjectIds []string `json:"project_ids" default:""`
+
+	AssignedTreatmentLogger AssignedTreatmentLoggerConfig `json:"assigned_treatment_logger"`
+	DeploymentConfig        DeploymentConfig              `json:"deployment_config"`
+	ManagementService       ManagementServiceConfig       `json:"management_service"`
+	MonitoringConfig        Monitoring                    `json:"monitoring_config"`
+	SwaggerConfig           SwaggerConfig                 `json:"swagger_config"`
+}
+
+type TreatmentServiceConfig struct {
+	Port       int      `json:"port" default:"8080"`
+	ProjectIds []string `json:"project_ids" default:""`
+
+	AssignedTreatmentLogger AssignedTreatmentLoggerConfig `json:"assigned_treatment_logger"`
+	DebugConfig             DebugConfig                   `json:"debug_config"`
+	NewRelicConfig          NewRelicConfig                `json:"new_relic_config"`
+	SentryConfig            SentryConfig                  `json:"sentry_config"`
+	DeploymentConfig        DeploymentConfig              `json:"deployment_config"`
+	PubSub                  PubSub                        `json:"pub_sub"`
+	ManagementService       ManagementServiceConfig       `json:"management_service"`
+	MonitoringConfig        Monitoring                    `json:"monitoring_config"`
+	SwaggerConfig           SwaggerConfig                 `json:"swagger_config"`
+	SegmenterConfig         map[string]interface{}        `json:"segmenter_config"`
+}
+
+type AssignedTreatmentLoggerConfig struct {
```

Comment:
I imagine that eventually the plugin will have a direct dependency on the treatment-service modules. Given this, shall we reuse the structs from `treatment-service/config/config.go` here too, instead of redefining them ?

Reply:
Yep I've replaced all these structs entirely with those from the treatment service. Can't remember why I copied them over manually but I remember having done it for some reason 😵‍💫